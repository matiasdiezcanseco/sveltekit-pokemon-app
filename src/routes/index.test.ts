import HomePage from './index.svelte'
import { beforeEach, describe, test, vi } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/svelte'

describe('HomePage', () => {
	beforeEach(() => {
		global.fetch = vi.fn().mockImplementation(() => {
			return Promise.resolve({
				json() {
					return Promise.resolve({
						name: 'Test Poke',
						height: 3,
						weight: 20,
						sprites: { front_default: '' }
					})
				}
			})
		})
	})

	test('should display inputed text', async () => {
		render(HomePage as any)
		const input = screen.getByRole('textbox')
		await fireEvent.input(input, { target: { value: 'mewtwo' } })
		expect(screen.getByText('mewtwo')).toBeTruthy()
	})

	test('should match snapshot ', () => {
		const wrapper = render(HomePage as any)
		expect(wrapper).toMatchSnapshot()
	})
})
