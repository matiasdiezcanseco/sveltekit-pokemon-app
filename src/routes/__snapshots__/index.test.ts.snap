// Vitest Snapshot v1

exports[`HomePage > should match snapshot  1`] = `
{
  "component": Routes$ {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        "",
        [],
        [],
        [Function],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {},
      "root": <div>
         
        <h1
          class="text-4xl text-center my-8 uppercase"
        >
          Svelte Kit Pokedex
        </h1>
         
        <input
          class="w-full rounded-md text-lg p-4 border-2 border-gray-200 mb-4"
          placeholder="Search Pokemon"
          type="text"
        />
         
        <div
          class="grid gap-4 grid-cols-1 md:grid-cols-2 "
        />
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
       
      <h1
        class="text-4xl text-center my-8 uppercase"
      >
        Svelte Kit Pokedex
      </h1>
       
      <input
        class="w-full rounded-md text-lg p-4 border-2 border-gray-200 mb-4"
        placeholder="Search Pokemon"
        type="text"
      />
       
      <div
        class="grid gap-4 grid-cols-1 md:grid-cols-2 "
      />
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
